### PR TITLE
Worldgen javadoc

### PR DESCRIPTION
**Pull request by jaskarth**
Javadoc for the new noise gen classes added in 21w05a, as well as surface builders.

### CODE REVIEW

File: mappings/net/minecraft/world/gen/surfacebuilder/SurfaceBuilder.mapping [hunk 1]
```diff
@@ -39,19 +40,33 @@ CLASS net/minecraft/class_3523 net/minecraft/world/gen/surfacebuilder/SurfaceBui
...
 		ARG 10 defaultFluid
+			COMMENT Default fluid of the chunk generator
 		ARG 11 seaLevel
+			COMMENT The sea level of the chunk generator
 		ARG 12 seed
+			COMMENT The world seed
 		ARG 14 surfaceBlocks
+			COMMENT The config passed to the chunk generator, stores the states used by the surface builder
 	METHOD method_15306 initSeed (J)V
+		COMMENT Runs before {@link generate} and allows for custom noise to be initialized.
```
Comment (Juuxel) [inline mappings/net/minecraft/world/gen/surfacebuilder/SurfaceBuilder.mapping]:
```suggestion
		COMMENT Runs before {@link #generate} and allows for custom noise to be initialized.
```
You might also need the signature here, not 100% sure (KDoc is much easier :tiny_potato:)

File: mappings/net/minecraft/world/gen/chunk/ChunkGenerator.mapping [hunk 2]
```diff
@@ -73,20 +75,24 @@ CLASS net/minecraft/class_2794 net/minecraft/world/gen/chunk/ChunkGenerator
...
+		ARG 3 heightmap
+		ARG 4 world
 	METHOD method_20402 getHeightOnGround (IILnet/minecraft/class_2902$class_2903;Lnet/minecraft/class_5539;)I
 		ARG 1 x
 		ARG 2 z
-		ARG 3 heightmapType
+		ARG 3 heightmap
+		ARG 4 world
 	METHOD method_26261 getColumnSample (IILnet/minecraft/class_5539;)Lnet/minecraft/class_4966;
+		COMMENT Returns a sample of all the blockstates in a column for use in structure generation.
```
Comment (Juuxel) [inline mappings/net/minecraft/world/gen/chunk/ChunkGenerator.mapping]:
```suggestion
		COMMENT Returns a sample of all the block states in a column for use in structure generation.
```

File: mappings/net/minecraft/world/gen/surfacebuilder/SurfaceBuilder.mapping [hunk 3]
```diff
@@ -39,19 +40,33 @@ CLASS net/minecraft/class_3523 net/minecraft/world/gen/surfacebuilder/SurfaceBui
...
 		ARG 2 chunk
+			COMMENT The current chunk being surface built
 		ARG 3 biome
+			COMMENT The biome in the column that is being surface built
 		ARG 4 x
+			COMMENT X coordinate of the column
 		ARG 5 z
+			COMMENT Z coordinate of the column
 		ARG 6 height
+			COMMENT Height of the column retrieved using {@link net.minecraft.world.Heightmap.Type.WORLD_SURFACE_WG}, and will never be lower than the sea level
```
Comment (Juuxel) [inline mappings/net/minecraft/world/gen/surfacebuilder/SurfaceBuilder.mapping]:
```suggestion
			COMMENT Height of the column retrieved using {@link net.minecraft.world.Heightmap.Type#WORLD_SURFACE_WG}, and will never be lower than the sea level
```

**Review by Juuxel — CHANGES_REQUESTED** [review]

**Review by YanisBft — APPROVED** [review]

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/world/gen/NoiseColumnSampler.mapping [hunk 4]
```diff
@@ -26,17 +28,22 @@ CLASS net/minecraft/class_5818 net/minecraft/world/gen/NoiseColumnSampler
 		ARG 7 islandNoise
 		ARG 8 densityNoise
 	METHOD method_33645 applySlides (DI)D
+		COMMENT Interpolates the noise at the top and bottom of the world.
 		ARG 1 noise
 		ARG 3 y
 	METHOD method_33646 getOffset (IDDD)D
+		COMMENT Calculates an offset for the noise.
+		COMMENT <p>For example in the overworld, this makes lower y values solid while making higher y values air.</p>
```
Comment (Juuxel) [inline mappings/net/minecraft/world/gen/NoiseColumnSampler.mapping:36]:
```suggestion
		COMMENT <p>For example in the overworld, this makes lower y values solid while making higher y values air.
```

File: mappings/net/minecraft/world/gen/StructureWeightSampler.mapping [hunk 5]
```diff
@@ -17,10 +19,13 @@ CLASS net/minecraft/class_5817 net/minecraft/world/gen/StructureWeightSampler
 	METHOD method_33640 ([F)V
 		ARG 0 array
 	METHOD method_33641 getStructureWeight (III)D
+		COMMENT Gets the structure weight from the array from the given position, or 0 if the position is out of bounds.
 		ARG 0 x
 		ARG 1 y
 		ARG 2 z
 	METHOD method_33642 calculateStructureWeight (III)D
+		COMMENT Calculates the structure weight for the given position.
+		COMMENT <p>The weight increases as x and z approach {@code (0, 0)}, and positive y values make the weight negative while negative y values make the weight positive.</p>
```
Comment (Juuxel) [inline mappings/net/minecraft/world/gen/StructureWeightSampler.mapping:28]:
```suggestion
		COMMENT <p>The weight increases as x and z approach {@code (0, 0)}, and positive y values make the weight negative while negative y values make the weight positive.
```

**Review by Juuxel — APPROVED** [review]
Closing `<p>` tags aren't needed but I'm fine either way.

**Review by LambdAurora — APPROVED** [review]

**Review by Mysterious-Dev — APPROVED** [review]

**Pull request merged**